opal/include/opal/sys/gcc_builtin/atomic.h

```diff
@@ -193,26 +193,30 @@ static inline intptr_t opal_atomic_swap_ptr(opal_atomic_intptr_t *addr, intptr_t
 
 static inline void opal_atomic_lock_init(opal_atomic_lock_t *lock, int32_t value)
 {
-    lock = value;
+    if (OPAL_ATOMIC_LOCK_UNLOCKED == value) {
```

Comment:
This either feels unnecessary or incomplete.  Either we need an atomic set here in both cases or neither case.